File: 1.Two_sum/Two_Sum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> dic = new HashMap<Integer, Integer>();
        int[] defaultResult = {0, 0};
        for(int i=0;i<nums.length;i++){
        	if(dic.get(target-nums[i]) != null){
        		int[] res = {dic.get(target-nums[i]), i+1};
        		return res;
        	}
        	else{
        		dic.put(nums[i], i+1);
        	}
        }
        return defaultResult;
    }
}

File: 1.Two_sum/Two_Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var dic = {};
    for(i=0;i<nums.length;i++){
    	if(target-nums[i] in dic){
    		return [dic[target-val], i+1];
    	}
    	else{
    		dic[nums[i]] = i+1;
    	}
    }
};